lib/index.ts

```diff
@@ -86,9 +94,26 @@ class PWMetrics {
         await sheets.appendResults(results.runs);
       }
     }
+
+    if (resultHasExpectationErrors && this.flags.expectations) {
+      throw new Error(messages.getMessage('HAS_EXPECTATION_ERRORS'));
+    }
+
     return results;
   }
 
+  resultHasExpectationErrors(metrics: MetricsResults): boolean {
+    const expectationErrors = metrics.timings.filter((timing: Timing) => {
+      const expectation = this.expectations[timing.id];
+      if (!expectation) {
+        return false;
+      }
+      const expectedErrorLimit = expectation.error;
+      return expectedErrorLimit !== undefined && timing.timing >= expectedErrorLimit;
```

Comment:
Take a look in [NormalizedExpectationMetrics](https://github.com/paulirish/pwmetrics/blob/5ed7d0f92cb294c23020638bc54160609f7cedad/types/types.ts#L110) interface and [where it's used in the code](https://github.com/paulirish/pwmetrics/pull/128/files#diff-8ad51fb63d3211deee9a805a512386adR23)